rpcs3/Input/hid_pad_handler.cpp

```diff
@@ -199,7 +199,11 @@ void hid_pad_handler<Device>::enumerate_devices()
 			}
 		}
 	}
-#else
+#endif
+
+	std::lock_guard lock(m_enumeration_mutex);
```

Comment:
This placement basically causes the entire enumeration to block the handler thread, which leads to hickups in the input.
That's the whole reason why this happens in a different thread in the first place.

Reply:
no. that's the part that's so slow lol

Reply:
It's because IOKit is not thread-safe, so basically hidapi should be single threaded on macOs

Reply:
> the comment you linked mentioned only MacOS but this issue also happens on Windows.

maybe there is two different issues

> It's because IOKit is not thread-safe, so basically hidapi should be single threaded on macOs
> I don't want any more custom commits to hidapi

I believe being slow is better than completely consistent and immediate crash when DualSense is connected but I understand you perhaps are considering complete replacement for the code path.